proposals/0259-DisabledSoftbuttons.md

```diff
@@ -69,13 +73,67 @@ Plus add an `isEnabled` flag to the `SubscribeButton` RPC.  If the button is set
         .
         .
         .
-            <param name="isEnabled" type="Boolean" mandatory="false">
-      <description>If true or missing, subscribed button must appear on screen (if applicable) and usable. If false, softbutton must appear on screen (if applicable) and not usable.  Button press events will still be sent to the app.
+            <param name="isDisabled" type="Boolean" mandatory="false">
+      <description>If false or missing, subscribed button must appear on screen and usable. If true, softbutton must appear on screen (if applicable) and not usable.  Button press events will still be sent to the app regardless.
       </description>
     </function>
     
 ```
 
+### Manager-Level Additions for the app developer
+
+#### Objective C
+1. Add a new soft button configuration to the screen manager:
+
+```ObjectiveC
+@property (strong, nonatomic, readonly) SDLSoftButtonConfiguration *softButtonConfiguration;
+```
+
+2. This will involve creating the soft button configuration:
+
+```ObjectiveC
+@interface SDLSoftButtonConfiguration: NSObject
+
+/// If true, and connected to a head unit that doesn't support disabled softbuttons, the soft button manager will not send soft button objects with the parameter isDisabled=false. If false, the soft buttons will be sent anyway but will appear enabled on the head unit. Defaults to true.
+@property (assign, nonatomic) BOOL disabledButtonFallback;
+
+@end
+```
+
+
+3. The `disabled` property will have to be added to `SDLSoftButtonState`:
+
+```ObjectiveC
+@property (assign, nonatomic, getter=isDisabled) BOOL disabled;
+```
+
+The soft button manager will then have to take both the configuration value and the `disabled` BOOL on the state into account to determine if the button will be sent or not and with what values. If on an older headunit that doesn't support this feature- the manager would make sure that the button is not sent to the head unit.
+
+
+#### Java
+1. Add a new soft button configuration to the screen manager:
+
+```Java
+    private SoftButtonConfiguration softButtonConfiguration;
```

Comment:
@bilal-alsharifi should this be private? It's developer-facing.

Reply:
@joeljfischer  I literally just copied the pattern set by [MenuConfiguration](https://github.com/smartdevicelink/sdl_java_suite/blob/93778656113f0023ec7025281b77e28000a1f5bd/base/src/main/java/com/smartdevicelink/managers/screen/menu/MenuConfiguration.java#L42)

Reply:
👍